scripts/src/javatests/org/oppia/android/scripts/common/MavenDependenciesListGeneratorTest.kt

```diff
@@ -0,0 +1,952 @@
+    )
+
+    val brokenLicenses = mavenDependenciesListGenerator.getAllBrokenLicenses(mavenDependenciesList)
+    assertThat(brokenLicenses).isEmpty()
+  }
+
+  @Test
+  fun testGetAllBrokenLicenses_multipleBrokenLicenses_returnsCorrectLicenseSet() {
+    val license1 = License.newBuilder().apply {
+      this.licenseName = "The Apache License, Version 2.0"
+      this.originalLink = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+    }.build()
+    val license2 = License.newBuilder().apply {
+      this.licenseName = "Simplified BSD License"
+      this.originalLink = "https://www.opensource.org/licenses/bsd-license"
+    }.build()
+    val mavenDependenciesList = listOf(
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_SCRAPABLE_LICENSE
+        this.artifactVersion = DATA_BINDING_VERSION
+        this.addLicense(license1)
+      }.build(),
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_NO_LICENSE
+        this.artifactVersion = PROTO_LITE_VERSION
+        this.addLicense(license2)
+      }.build()
+    )
+
+    val brokenLicenses = mavenDependenciesListGenerator.getAllBrokenLicenses(mavenDependenciesList)
+    assertThat(brokenLicenses.size).isEqualTo(2)
+    verifyLicenseHasVerifiedLinkNotSet(
+      license = brokenLicenses.elementAt(0),
+      licenseName = "The Apache License, Version 2.0",
+      originalLink = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+    )
+    verifyLicenseHasVerifiedLinkNotSet(
+      license = brokenLicenses.elementAt(1),
+      licenseName = "Simplified BSD License",
+      originalLink = "https://www.opensource.org/licenses/bsd-license"
+    )
+  }
+
+  @Test
+  fun testGetAllBrokenLicenses_licenseOriginalLinkInvalid_notPresentInReturnedSet() {
+    val license1 = License.newBuilder().apply {
+      this.licenseName = "The Apache License, Version 2.0"
+      this.originalLink = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+      this.scrapableLink = ScrapableLink.newBuilder().apply {
+        url = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+      }.build()
+    }.build()
+    val license2 = License.newBuilder().apply {
+      this.licenseName = "Crashlytics Terms of Service"
+      this.originalLink = "https://try.crashlytics.com/terms/terms-of-service.pdf"
+      this.isOriginalLinkInvalid = true
+    }.build()
+    val mavenDependenciesList = listOf(
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_SCRAPABLE_LICENSE
+        this.artifactVersion = DATA_BINDING_VERSION
+        this.addLicense(license1)
+      }.build(),
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_NO_LICENSE
+        this.artifactVersion = PROTO_LITE_VERSION
+        this.addLicense(license2)
+      }.build()
+    )
+
+    val brokenLicenses = mavenDependenciesListGenerator.getAllBrokenLicenses(mavenDependenciesList)
+    assertThat(brokenLicenses).doesNotContain(license2)
+  }
+
```

Comment:
I will add KDocs for all the public methods and more test cases for the remaining functions tomorrow.
@BenHenning Until then, please do let me know if the PR looks good overall.